masterpdfeditor4: init at 4.3.89

eduardosm opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

The free version of masterpdfeditor 5 adds a watermark to saved PDFs, whereas version 4 does not.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

SuperSandro2000 commented on pkgs/applications/misc/masterpdfeditor4/default.nix:
```diff
@@ -0,0 +1,45 @@
+
+  buildInputs = [ qtbase qtsvg sane-backends stdenv.cc.cc ];
+
+  dontStrip = true;
```
Why are we not stripping it?

eduardosm replied:
That's how it is done in the original masterpdfeditor package.

SuperSandro2000 replied:
If there is no reason to not strip like crashes we want to do this.

eduardosm replied:
Seems to be working, so I removed the `dontStrip`.

SuperSandro2000 commented on pkgs/applications/misc/masterpdfeditor4/default.nix:
```diff
@@ -0,0 +1,45 @@
+  installPhase = ''
+    runHook preInstall
+
+    p=$out/opt/masterpdfeditor4
```
p? Please use a longer name.

eduardosm replied:
The original masterpdfeditor has this, but I renamed it.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108221` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>masterpdfeditor4</li>
</ul>
</details>

eduardosm committed:
masterpdfeditor4: init at 4.3.89

The free version of masterpdfeditor 5 adds a watermark to saved PDFs, whereas version 4 does not.

SuperSandro2000 commented:
Please do not rebase on master except if it is necessary.
